packages/dashboard/src/components/robots/robot-page.tsx

```diff
@@ -1,31 +1,81 @@
 /* istanbul ignore file */
 
-import { makeStyles } from '@material-ui/core';
+import { makeStyles, Grid, Card } from '@material-ui/core';
 import React from 'react';
+import * as RmfModels from 'rmf-models';
+import { Fleet } from 'api-client';
 import { RobotPanel, VerboseRobot } from 'react-components';
-import { RmfIngressContext } from '../rmf-app';
+import { LeafletContext } from 'react-leaflet';
+import {
+  BuildingMapContext,
+  RmfIngressContext,
+  DispensersContext,
+  IngestorsContext,
+} from '../rmf-app';
+import ScheduleVisualizer from '../schedule-visualizer';
+import {
+  GetFleets,
+  SubscribeFleet,
+  SubscribeDispenser,
+  SubscribeIngestor,
+} from '../../util/common-subscriptions';
+
+const UpdateRate = 1000;
 
 const useStyles = makeStyles((theme) => ({
   robotPanel: {
+    height: '100%',
+  },
+  container: {
     padding: `${theme.spacing(4)}px`,
     height: '100%',
     backgroundColor: theme.palette.background.default,
   },
+  mapPanel: {
+    height: '100%',
+    marginRight: theme.spacing(2),
+    flex: '1 0 auto',
+  },
 }));
 
 export function RobotPage() {
   const classes = useStyles();
-  const { fleetsApi } = React.useContext(RmfIngressContext) || {};
+  const rmfIngress = React.useContext(RmfIngressContext);
+  const sioClient = React.useContext(RmfIngressContext)?.sioClient;
+  const buildingMap = React.useContext(BuildingMapContext);
+
+  const [leafletMap, setLeafletMap] = React.useState<LeafletContext>({});
+  const [_triggerRender, setTriggerRender] = React.useState(0); // eslint-disable-line @typescript-eslint/no-unused-vars
+  React.useEffect(() => {
+    const interval = setInterval(() => setTriggerRender((prev) => prev + 1), UpdateRate);
+    return () => clearInterval(interval);
+  }, []);
+
+  // get work cells to display on map
+  const dispensers = React.useContext(DispensersContext);
+  const dispenserStatesRef = React.useRef<Record<string, RmfModels.DispenserState>>({});
+  SubscribeDispenser(sioClient, dispensers, dispenserStatesRef);
+
+  const ingestors = React.useContext(IngestorsContext);
+  const ingestorStatesRef = React.useRef<Record<string, RmfModels.IngestorState>>({});
+  SubscribeIngestor(sioClient, ingestors, ingestorStatesRef);
+
+  // schedule visualizer fleet
+  const [fleets, setFleets] = React.useState<Fleet[]>([]);
+  GetFleets(rmfIngress, setFleets);
+  const fleetStatesRef = React.useRef<Record<string, RmfModels.FleetState>>({});
+  SubscribeFleet(sioClient, fleets, fleetStatesRef);
 
+  // robot panel stuff
   const [hasMore, setHasMore] = React.useState(true);
   const [page, setPage] = React.useState(0);
   const [verboseRobots, setVerboseRobots] = React.useState<VerboseRobot[]>([]);
   const fetchVerboseRobots = React.useCallback(async () => {
```

Comment:
would it help if the fetching of the robots states can be toggled with `autorefresh` like the tasks tabs to fix the problem of the position og the centering feature?

Reply:
Sure! Maybe I could create a separate task for this.

Reply:
Fixed the positioning issue by calling `fetchVerboseRobots` when clicking the robot.